### ISSUE TITLE
Ensure API Express metrics are using correct protocol (http or https)

### ISSUE THREAD

**Issue by mrjones-plip** [medic/cht-watchdog#128]
A production instance recently had port `80` accidentally stop working.  As it turns out, API Express metrics always defaults to `80` and then gets redirected to port `443` where the metrics actually are served.  Without the redirect, the metrics fail. 

Related:
* https://github.com/medic/cht-watchdog/pull/74
* https://github.com/medic/cht-watchdog/pull/75
* https://github.com/medic/cht-docs/issues/1145
* https://github.com/medic/cht-core/pull/8354

Checking production watchdog instance that Medic runs, I see all requests are hitting port `80` first:

![Image](https://github.com/user-attachments/assets/84b1eff8-1166-45ef-a13d-7da0e3136ecb)

**Comment by mrjones-plip** [comment]
tl;dr - i propose we force scrapes of express metrics to be over TLS. I've added a PR to this extent.

---

Soooo - this is tricky.  On a local docker helper instance I had both port  `10091` running with `http` and port `10454` running `https`.  By default express metrics runs a regex and drops the `https` (or `http`) used in `cht-instances.yml` - check out the `regex` [line](https://github.com/medic/cht-watchdog/blob/62a7be1e39d9513ed9a174b8d641b3282344a5f5/prometheus/config/prometheus.yml#L19):

```yaml
  - job_name: cht-express-metrics
    metrics_path: /api/v1/express-metrics
    file_sd_configs:
      - files:
        - '/etc/prometheus/cht-instances.yml'
    relabel_configs:
    - source_labels: [__address__]
      regex: "(?:https?:\\/\\/|)(?:www\\.|)(.*?)(?:\\/|)$"
      target_label: instance
      replacement: "$1"
    - source_labels: [instance]
      target_label: __address__
```

So what happens in this default scenario is
1.  watchdog reads in the URL from `cht-instances.yml` as `https://192-168-68-26.local-ip.medicmobile.org:10454`
2. then regex above drops the `https` and makes this URL: `http://192-168-68-26.local-ip.medicmobile.org:10454/api/v1/express-metrics`
3. it does a scrape and of course errors out: `server returned HTTP status 497`  - translation - bruh - i can't speak `http` to an `https` server

The fix is one of these then:
1. either we update the above scrape [config file](https://github.com/medic/cht-watchdog/blob/62a7be1e39d9513ed9a174b8d641b3282344a5f5/prometheus/config/prometheus.yml) and add a `scheme: "https"` - which forces ALL requests to go to `https` - OR `
2. um...there is second one now that I think about it! I was going to say you can put in `http://192-168-68-26.local-ip.medicmobile.org:10091` in the `cht-instances.yml` file - but then in dev I had to comment out the redirect we have on the `http` port add and add the `server.conf` include and then it could get both the JSON monitoring endpoint AND the express metrics.  but this is so not real world that it's useless

by going with option 1 (`scheme: "https"` ) we're breaking it in dev environment for not docker helper, but it will work in docker helper, which seems common enough.

i'll see where I get with the PR!

**Comment by mrjones-plip** [comment]
cc @jkuester - but I'm tapping Kenn for the PR review!
